models/sourcefile.py

```diff
@@ -175,7 +175,10 @@ def make_oembed_url(url):
 
         oembed_url = None
         if url_parsed.hostname.lower() in ['youtube.com', 'www.youtube.com', 'youtu.be']:
-            to_url = 'https://%s%s?%s' % (url_parsed.hostname, url_parsed.path, url_parsed.query)
+            # We want to strip any `si` value from the query, as it can be used to track the original
+            # YouTube user that shared the link.
+            query = re.sub(r'[?&]si=[^&?]*', '', url_parsed.query)
+            to_url = 'https://%s%s?%s' % (url_parsed.hostname, url_parsed.path, query)
```

Comment:
I think I prefer this approach. I'm not keen on fashioning regex and constructing URLs when there is existing code to do that. I know the implementation has been doing that, but not a good practice, IMHO.

```python
query_params = parse_qs(url_parsed.query)
if "si" in query_params:
    del query_params["si"]
    url_parsed = url_parsed._replace(
        query=urlencode(query_params, doseq=True))
```

Reply:
@bradchoate Done.